packages/delegation-toolkit/src/DelegationFramework/ERC20StreamingEnforcer/methods/getAvailableAmount.ts

```diff
@@ -0,0 +1,28 @@
+import { ERC20StreamingEnforcer } from '@metamask/delegation-abis';
+import type { Address, Client, Hex } from 'viem';
+import { readContract } from 'viem/actions';
+
+export type ReadGetAvailableAmountParameters = {
+  client: Client;
+  contractAddress: Address;
+  delegationManager: Address;
+  delegationHash: Hex;
+};
+
+export const read = async ({
+  client,
+  contractAddress,
+  delegationManager,
+  delegationHash,
+}: ReadGetAvailableAmountParameters) => {
+  const result = await readContract(client, {
+    address: contractAddress,
+    abi: ERC20StreamingEnforcer.abi,
+    functionName: 'getAvailableAmount',
+    args: [delegationManager, delegationHash],
+  });
+
+  return {
+    availableAmount: result,
```

Comment:
nit: rename the return parameter to use shorthand property name 🎉 

```suggestion
  const availableAmount = await readContract(client, {
    address: contractAddress,
    abi: ERC20StreamingEnforcer.abi,
    functionName: 'getAvailableAmount',
    args: [delegationManager, delegationHash],
  });

  return {
    availableAmount,
```